src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt

```diff
@@ -133,7 +136,23 @@ class ApplicationsController(
 
     val serializedData = objectMapper.writeValueAsString(body.data)
 
-    val applicationResult = applicationService.updateApplication(applicationId, serializedData, username)
+    val applicationResult = when(body) {
+      is UpdateApprovedPremisesApplication -> applicationService.updateApprovedPremisesApplication(
```

Comment:
Are all these required? We won't necessarily have all these things at the start of the application

Reply:
`isWomensApplication`, `isPipeApplication`, `releaseType` and `arivalDate` are all optional in the OpenAPI spec and in the service layer so the FE is fine to just send whatever it has

Reply:
Cool cool 👍